File: dsp12.pas
(****  PROGRAM NO.12   F-RESPONSE OF MOVING-AVERAGE FILTER  ****)
program DSP12;
uses graph3;
var J,W,H: real;
var N,M,K: integer;
begin
  writeln('Enter value of M');
  readln(M);
  J:=1/(2*M+1);
  graphmode;
  for N:=1 to 320 do
    begin
      W:=3.141593*(N-1)/320;H:=J;
        for K:=1 to M do
          begin
            H:=H+2*J*cos(K*W);
          end;
        draw(N,100,N,round(100-50*abs(H)),1);
    end;
end.
(***************************************************************)

File: dsp12.c
#include <graph.h>
#include <conio.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

void main (void)
    {
    /*
     *** PROGRAM NO.12  F-RESPONSE OF MOVING-AVERAGE FILTER ****
     */
    int    iCount, iCount2, iM;
    float  fH, fJ, fW;

    _clearscreen  (_GCLEARSCREEN);
    printf ("PROGRAM 12: F-response of moving-average filter\n");

    printf ("Enter (integer) value of M :");
    scanf ("%d", &iM);

    fJ = 1.0 / (2.0 * (float)iM + 1.0);

    _setvideomode( _MRES16COLOR );
    
    for (iCount = 1; iCount <= 320; iCount++)
        {
        fW = 3.141593 * (float)(iCount - 1) / 320.0;
        fH = fJ;
        for (iCount2 = 1; iCount2 <= iM; iCount2++)
            fH = fH + 2.0 * fJ * cos ((float)iCount2 * fW);

        _moveto (iCount, 100);
        _lineto (iCount, (int)(100 - 50.0 * fabs (fH) + 0.5));
        }

    getch();
    _setvideomode( _DEFAULTMODE );

    /*
     ***********************************************************
     */
    }
